carbon-design-system/carbon-for-ibm-dotcom
# feat(feature-section): add "Feature section" layout component

### Related Ticket(s)
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6246

DDS Tickets: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6118
HC JIRA Ticket: https://jsw.ibm.com/browse/HC-1971

### Description
- This PR adds the "Feature Section" layout component/pattern that's needed in the new "Service" landing page template that will be implemented in the AEM CMS.
- This component is a variant of HC's [Module 11A](https://pages.github.ibm.com/MSC-Cloud/hc_fractal/components/detail/11a--default.html) and this iteration is the first version that will be going into the MVE state of the "Service" landing page.

### Design & Functional Specs
- https://github.com/carbon-design-system/carbon-for-ibm-dotcom-website/wiki/Feature-section

![component--feature-card-block-large--visual-specs--max copy](https://user-images.githubusercontent.com/1815714/121596789-ea58b600-ca0d-11eb-9789-aaa4be7e7d58.jpg)


### Changelog

**New Components/Props**

- `<dds-feature-section>` – this new layout component is a variant/extension of the `<dds-feature-card>`.
- `<dds-feature-section-card-link>` – this component is an extension of the `<dds-card-link>` and needed because the feature section component requires a card-link with higher max-width.
- `media-alignment` - a prop  `(right (default) | left)` that can be passed as an attribute.

**Removed**

- N/A

**Note**

- The current implementation of the `<dds-card-link>` UI component doesn't offer CTA types such as Video, document, external, etc. The component looks to be extended from the `dds-card`, where it should be using `dds-cta-card` component.
- Looks like this is a known issue (see - https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5740)


**Testing Instructions**
- Visit the `Feature section` - `/?path=/story/components-feature-section--default`
- Ensure that the component matches with the design specs for all breakpoints.
